src/library/traktor/traktorfeature.cpp

```diff
@@ -50,6 +50,82 @@ bool TraktorPlaylistModel::isColumnHiddenByDefault(int column) {
     return BaseSqlTableModel::isColumnHiddenByDefault(column);
 }
 
+void addCuesToTrack(CueDAO &cueDao, TrackPointer pTrack, QString traktorTrackId, const QSqlDatabase &m_database) {
+    QList<CuePointer> cues;
+    // A hash from hotcue index to cue id and cue*, used to detect if more
+    // than one cue has been assigned to a single hotcue id.
+    QMap<int, QPair<int, CuePointer> > dupe_hotcues;
+
+    const double posMultiplier = double(pTrack->getSampleRate())/48000.0; //Assumed 48k in xml parse
+
+    QSqlQuery query(m_database);
+    query.prepare("SELECT * FROM traktor_cues WHERE track_id = :track_id");
+    query.bindValue(":track_id", traktorTrackId);
+    if (query.exec()) {
+        QSqlRecord record = query.record();
+        const int idColumn = record.indexOf("id");
+        const int trackIdColumn = record.indexOf("track_id");
+        const int typeColumn = record.indexOf("type");
+        const int positionColumn = record.indexOf("position");
+        const int lengthColumn = record.indexOf("length");
+        const int hotcueColumn = record.indexOf("hotcue");
+        const int labelColumn = record.indexOf("label");
+        const int colorColumn = record.indexOf("color");
+
+        while (query.next()) {
+            int id = query.value(idColumn).toInt();
+            TrackId trackId(query.value(trackIdColumn));
+            int type = query.value(typeColumn).toInt();
+            int position = int(double(query.value(positionColumn).toInt()) * posMultiplier);
+            int length = query.value(lengthColumn).toInt();
+            int hotcue = query.value(hotcueColumn).toInt();
+            QString label = query.value(labelColumn).toString();
+            QColor color = QColor::fromRgba(query.value(colorColumn).toInt());
+
+            CuePointer pCue(new Cue(-1, trackId, (Cue::CueType)type,
+                       position, length, hotcue, label, color));
+            if (hotcue != -1) {
+                if (dupe_hotcues.contains(hotcue)) {
+                    cues.removeOne(dupe_hotcues[hotcue].second);
+                }
+                dupe_hotcues[hotcue] = qMakePair(id, pCue);
+            }
+            cues.push_back(pCue);
+        }
+        qDebug() << "Found " << cues.count() << " cues";
+        cueDao.saveTrackCues(pTrack->getId(), cues);
+        pTrack->setCuePoints(cues); //needed?
+    } else {
+        LOG_FAILED_QUERY(query);
+    }
+}
+
+TrackPointer TraktorTrackModel::getTrack(const QModelIndex& index) const {
+    QString id = index.sibling(index.row(), fieldIndex("id")).data().toString();
+    QString location = index.sibling(index.row(), fieldIndex("location")).data().toString();
+    bool track_already_in_library = m_pTrackCollection->getTrackDAO().trackExistsInDatabase(location);
+    TrackPointer pTrack = BaseExternalTrackModel::getTrack(index);
+
+    if (pTrack && !track_already_in_library) {
```

Comment:
This is the check that has to be removed (besides other minor changes) to be able to import changed Traktor cues.